lib/qa/authorities/linked_data/search_query.rb

```diff
@@ -0,0 +1,122 @@
+        end
+
+        def preds_for_search
+          { required: required_search_preds, optional: optional_search_preds }
+        end
+
+        def required_search_preds
+          label_pred_uri = auth_config.search_results_label_predicate
+          raise Qa::InvalidConfiguration, "required label_predicate is missing in search configuration for LOD authority #{auth_name}" if label_pred_uri.nil?
+          { label: label_pred_uri }
+        end
+
+        def optional_search_preds
+          preds = {}
+          preds[:altlabel] = auth_config.search_results_altlabel_predicate unless auth_config.search_results_altlabel_predicate.nil?
+          preds[:id] = auth_config.search_results_id_predicate unless auth_config.search_results_id_predicate.nil?
+          preds[:sort] = auth_config.search_results_sort_predicate unless auth_config.search_results_sort_predicate.nil?
+          preds
+        end
+
+        def consolidate_search_results(results)
+          consolidated_results = {}
+          return consolidated_results if results.nil? || !results.count.positive?
+          results.each do |statement|
+            stmt_hash = statement.to_h
+            uri = stmt_hash[:uri].to_s
+            consolidated_hash = init_consolidated_hash(consolidated_results, uri, stmt_hash[:id].to_s)
+
+            consolidated_hash[:label] = object_value(stmt_hash, consolidated_hash, :label, false)
+            consolidated_hash[:altlabel] = object_value(stmt_hash, consolidated_hash, :altlabel, false)
+            consolidated_hash[:sort] = object_value(stmt_hash, consolidated_hash, :sort, false)
+            consolidated_results[uri] = consolidated_hash
+          end
+          consolidated_results.each do |res|
+            consolidated_hash = res[1]
+            consolidated_hash[:label] = sort_string_by_language consolidated_hash[:label]
+            consolidated_hash[:altlabel] = sort_string_by_language consolidated_hash[:altlabel]
+            consolidated_hash[:sort] = sort_string_by_language consolidated_hash[:sort]
+          end
+          consolidated_results
+        end
+
+        def convert_search_to_json(consolidated_results)
+          json_results = []
+          consolidated_results.each do |uri, h|
+            json_results << { uri: uri, id: h[:id], label: full_label(h[:label], h[:altlabel]), sort: h[:sort] }
+          end
+          json_results
+        end
+
+        def full_label(label = [], altlabel = [])
+          lbl = wrap_labels(label)
+          lbl += " (#{altlabel.join(', ')})" unless altlabel.nil? || altlabel.length <= 0
+          lbl = lbl.slice(0..95) + '...' if lbl.length > 98
+          lbl.strip
+        end
+
+        def wrap_labels(labels)
+          lbl = "" if labels.nil? || labels.size.zero?
+          lbl = labels.join(', ') if labels.size.positive?
+          lbl = '[' + lbl + ']' if labels.size > 1
+          lbl
+        end
+
+        def sort_search_results(json_results)
```

Comment:
I agree that the sorting code was complex.  I moved the processing of special cases into separate methods and added comments.  Take a look and let me know if this addresses your concerns.

See [search_query.rb#sort_search_results](https://github.com/projecthydra-labs/questioning_authority/blob/linked_data/lib/qa/authorities/linked_data/search_query.rb#L97-#L129)
See [sort_search_results rspec tests](https://github.com/projecthydra-labs/questioning_authority/blob/linked_data/spec/lib/authorities/linked_data/search_query_spec.rb#L201-L271)